tests/MongoDB.Driver.Examples/DocumentationExamples.cs

```diff
@@ -1246,6 +1248,324 @@ public void Example_58()
+            //{
+            //    $sort: { "numberSold": 1 }
+            //}])
+
+            // Start Aggregation Example 2
+            var pipeline = new EmptyPipelineDefinition<BsonDocument>()
+                .Unwind("items")
+                .Match(Builders<BsonDocument>.Filter.Eq("items.fruit", "banana"))
+                .Group(new BsonDocument
+                {
+                    { "_id", new BsonDocument("day", new BsonDocument("$dayOfWeek", "$date")) },
+                    { "count", new BsonDocument("$sum", "$items.quantity") },
+                })
+                .Project(new BsonDocument
+                {
+                    { "dayOfWeek", "$_id.day" },
+                    { "numberSold", "$count" },
+                    { "_id", 0 }
+                })
+                .Sort(Builders<BsonDocument>.Sort.Ascending("numberSold"));
+
+            var cursor = collection.Aggregate(pipeline);
+            // End Aggregation Example 2
+
+            Render(pipeline).Should().Be(@"
+                [{
+                    $unwind : '$items'
+                },
+                {
+                    $match : { 'items.fruit' : 'banana' }
+                },
+                {
+                    $group : {
+                        _id : { day : { $dayOfWeek : '$date' } },
+                        count : { $sum : '$items.quantity' }
+                    }
+                },
+                {
+                    $project : { dayOfWeek : '$_id.day', numberSold : '$count', _id : 0 }
+                },
+                {
+                    $sort : { 'numberSold' : 1 }
+                }]");
+        }
+
+        [Fact]
+        public void Aggregation_Example_3()
+        {
+            RequireServer.Check().Supports(Feature.Aggregate);
+
+            //db.sales.aggregate([
+            //{
+            //    $unwind: "$items"
+            //},
+            //{
+            //    $group: {
+            //        _id: { day: { $dayOfWeek: "$date" } },
+            //        items_sold: { $sum: "$items.quantity" },
+            //        revenue: { $sum: { $multiply: ["$items.quantity", "$items.price"] } }
+            //    }
+            //},
+            //{
+            //    $project: { day: "$_id.day", revenue: 1, items_sold: 1, 
+            //                discount: { $cond: { if : { $lte: ["$revenue", 250] }, then: 25, else : 0 }}
+            //    }
+            //}])
+
+            // Start Aggregation Example 3
+            var pipeline = new EmptyPipelineDefinition<BsonDocument>()
+                .Unwind("items")
+                .Group(new BsonDocument
+                {
+                    { "_id", new BsonDocument("day", new BsonDocument("$dayOfWeek", "$date")) },
+                    { "items_sold", new BsonDocument("$sum", "$items.quantity") },
+                    { "revenue", new BsonDocument("$sum", new BsonDocument("$multiply", new BsonArray(new [] { "$items.quantity", "$items.price" }))) }
```

Comment:
An alternative (perhaps nicer?) way to populate a BsonArray using collection initializer syntax is:

    new BsonArray { "$items.quantity", "$items.price" }

Reply:
Same comment for a few other places where a new BsonArray is created.

Reply:
Agree, look much nicer